test/hotspot/jtreg/compiler/c2/irTests/scalarReplacement/AllocationMergesNestedPhiTests.java

```diff
@@ -0,0 +1,742 @@
+          p2 = new Point(x, y);
+
+        try {
+            if (cond1 && cond2)
+                throw new Exception();
+        } catch (Exception e) {}
+
+        return p2.getX();
+    }
+
+    //--------------------------------------------------------------------------------------------------------------------------------------------
+
+    @DontCompile
+    int testNestedPhiPolymorphic_Interp(boolean cond1, boolean cond2, int x, int y) { return testNestedPhiPolymorphic(cond1, cond2, x, y); }
+
+    @Test
+    @IR(counts = { IRNode.ALLOC, "3" },
+        phase = CompilePhase.PHASEIDEAL_BEFORE_EA,
+        applyIfPlatform = {"64-bit", "true"},
+        applyIf = {"EliminateAllocations", "true"})
+    @IR(counts = { IRNode.ALLOC, "0" },
+        phase = CompilePhase.ITER_GVN_AFTER_ELIMINATION,
+        applyIfPlatform = {"64-bit", "true"},
+        applyIf = {"EliminateAllocations", "true"})
+    int testNestedPhiPolymorphic_C2(boolean cond1, boolean cond2, int x, int y) { return testNestedPhiPolymorphic(cond1, cond2, x, y); }
+
+    @ForceInline
+    int testNestedPhiPolymorphic(boolean cond1, boolean cond2, int x, int l) {
+        Shape obj1 = new Square(l);
+        if (cond1)
+            obj1 = new Circle(l);
+        Shape obj2 = obj1;
+        if (cond2)
+            obj2 = new Circle(x + l/5);
+        return obj2.l;
+    }
+
+    //--------------------------------------------------------------------------------------------------------------------------------------------
+
+    @DontCompile
+    int testNestedPhiWithTrap_Interp(boolean cond1, boolean cond2, int x, int y) { return testNestedPhiWithTrap(cond1, cond2, x, y); }
+
+    @Test
+    @IR(counts = { IRNode.ALLOC, "3" },
+        phase = CompilePhase.PHASEIDEAL_BEFORE_EA,
+        applyIfPlatform = {"64-bit", "true"},
+        applyIf = {"EliminateAllocations", "true"})
+    @IR(counts = { IRNode.ALLOC, "2" },
+        phase = CompilePhase.ITER_GVN_AFTER_ELIMINATION,
+        applyIfPlatform = {"64-bit", "true"},
+        applyIf = {"EliminateAllocations", "true"})
+    int testNestedPhiWithTrap_C2(boolean cond1, boolean cond2, int x, int y) { return testNestedPhiWithTrap(cond1, cond2, x, y); }
+
+    @ForceInline
+    int testNestedPhiWithTrap(boolean cond1, boolean cond2, int x, int y) {
+        Point p1 = new Point(x, y);
+        if (cond1)
+            p1 = new Point(x, y);
+
+        Point p2 = p1;
+        dummy();
+        if (cond2)
+            p2 = new Point(x, y);
+
+        return p2.x;
+    }
+
+    //--------------------------------------------------------------------------------------------------------------------------------------------
+
+   @DontCompile
+    int testNestedPhiWithLambda_Interp(boolean cond1, boolean cond2, int x, int y) { return testNestedPhiWithLambda(cond1, cond2, x, y); }
+
+    @Test
+    @IR(counts = { IRNode.ALLOC, ">=3" },
+        phase = CompilePhase.PHASEIDEAL_BEFORE_EA,
+        applyIfPlatform = {"64-bit", "true"},
```

Comment:
Looks like all tests should be run on 64bit platform? If so perhaps you can just add the requirement at the top of this file, in the annotations section.

Reply:
@JohnTortugo I'd generally advise against restricting files to limited platforms. The tests can still run on other platforms, and possibly find bugs there. Limiting IR rules allows us to make strong assertions on limited platforms (e.g. asserting that some optimizations are done), while at least testing weaker properties (correctness) on all platforms.

Reply:
I thought the test would not be run if no annotation was applicable to the current platform.
Anyway, if you think this is the best way, that's fine for me too.

Reply:
@JohnTortugo This is a very important distinction: the IR `applyIf` only restricts the IR rule, not the running of the test itself ;)

Reply:
As I said, I see your point: you want to run the test not to verify the IR, but to compare the output of the method when running in interpreted and in compiled mode. but this is still arguably confusing. That being said, my confusion was that, AFAIU, the "IR" tests are for testing IR patterns. If there is no IR validation on a "IR test" method so you'd imagine this is not really an "IR test", its a "regular JTREG" test.